### ISSUE TITLE
uploading JS libraries

### ISSUE THREAD

**Issue by shiffman** [processing/p5.js-web-editor#196]
I'm not able to upload JS libraries at the moment.  This has worked for me in the past.  I've tried with [RiTa.js](https://github.com/dhowe/RiTaJS/blob/master/dist/rita.js) and JQuery.   The weird thing is that if I upload my own JS file it seems to not have an issue.  Here's an example where I uploaded both`particle.js` and `jquery-3.1.1.min.js`.

http://alpha.editor.p5js.org/projects/S1lPBnXZx

JQuery is not recognized but `Particle` is.

One clue is I can't view the contents of the jquery file in the editor.  However, when I download the project as a zip it is, in fact, there.

cc @aparrish

**Comment by catarak** [comment]
The jQuery library is actually uploaded correctly here--it's just hosted on AWS rather than in the MongoDB. For very large text files (JS, CSS, CSV, JSON, etc) I decided that it made more sense to do it this way because they were actually too big for the CodeMirror editor to handle, and it would become very sluggish. I'm not sure how to make it clear that this is the case though!

The preview is, however, not injecting the jQuery file, which is definitely a bug.

**Comment by shiffman** [comment]
Yes, that makes sense!  Maybe we should provide some sort of warning where the code would normally appear like "this JS (or otherwise) file is too large to edit" when you click on the file in the editor?  And then this ultimately relates to #4.

**Comment by shiffman** [comment]
Also, is this then the same issue as #165?

**Comment by catarak** [comment]
This is also related to #57. 

This is not the same issue as #165, but definitely related. I've been working on that issue locally and refactoring/implementing better the preview generation code, but haven't finished it yet.
